www/class/centreonContact.class.php

```diff
@@ -249,6 +249,136 @@ public function getObjectForSelect2($values = array(), $options = array())
         return $items;
     }
 
+    /**
+     * Check if a password respects configured policy
+     *
+     * @param string $password
+     * @param int|null $contactId
+     * @return bool
+     * @throws \Exception
+     */
+    public function respectPasswordPolicyOrFail(string $password, ?int $contactId): void
+    {
+        $result = $this->db->query("SELECT * from password_security_policy");
+        $passwordPolicy = $result->fetch(\PDO::FETCH_ASSOC);
+        if ($passwordPolicy === false) {
+            throw new \Exception('Password security policy not found');
+        }
+
+        $this->respectPasswordCharactersOrFail($passwordPolicy, $password);
+
+        if ($contactId !== null) {
+            $this->respectPasswordChangePolicyOrFail($passwordPolicy, $password, $contactId);
+        }
+    }
+
+    /**
+     * Check if a password respects configured policy about characters (length, special characters, ...)
+     *
+     * @param array<string,mixed> $passwordPolicy
+     * @param string $password
+     * @return void
+     * @throws \Exception
+     */
+    private function respectPasswordCharactersOrFail(array $passwordPolicy, string $password): void
+    {
+        $doesRespectPassword = true;
+
+        $errorMessage = sprintf(
+            _("Your password must be %d characters long"),
+            (int) $passwordPolicy['password_length']
+        );
+        if (strlen($password) < (int) $passwordPolicy['password_length']) {
+            $doesRespectPassword = false;
+        }
+
+        $characterRules = [
+            'uppercase_characters' => [
+                'pattern' => '/[A-Z]/',
+                'error_message' =>  _("uppercase characters"),
+            ],
+            'lowercase_characters' => [
+                'pattern' => '/[a-z]/',
+                'error_message' =>  _("lowercase characters"),
+            ],
+            'integer_characters' => [
+                'pattern' => '/[0-9]/',
+                'error_message' =>  _("integer characters"),
+            ],
+            'special_characters' => [
+                'pattern' => '/[@$!%*?&]/',
```

Comment:
if the branch is rebase on password-security-policy branch you can use the constant SecurityPolicy::SPECIAL_CHARACTERS_LIST